tfhe/src/shortint/engine/server_side/scalar_sub.rs

```diff
@@ -51,14 +51,17 @@ impl ShortintEngine {
         ct: &mut Ciphertext,
         scalar: u8,
     ) -> EngineResult<()> {
-        let modulus = server_key.message_modulus.0 as u64;
         // Direct scalar computation is possible
         if server_key.is_scalar_sub_possible(ct, scalar) {
             self.unchecked_scalar_sub_assign(ct, scalar)?;
         } else {
             let scalar = u64::from(scalar);
-            // If the scalar is too large, PBS is used to compute the scalar mul
-            let acc = self.generate_lookup_table(server_key, |x| (x - scalar) % modulus)?;
+            // If the scalar is too large, PBS is used to compute the scalar sub
+            let acc = self.generate_msg_lookup_table(
+                server_key,
+                |x| x.wrapping_sub(scalar),
+                server_key.message_modulus,
+            )?;
```

Comment:
is this still correct when x is replaced by x % modulus ?

Reply:
Yes, I think so

But there may be a problem with wrapping if `ct_modulus % message_modulus != 0`
I think `x + msg_modulus - (scalar % msg_modulus)` is what we want. Does it make sense to you?

Reply:
indeed for non power of two it's probably better the way you just wrote it

Reply:
let's go with that formula then I'd say

Reply:
Done